py-polars/polars/io/delta.py

```diff
@@ -1,9 +1,12 @@
 from __future__ import annotations
 
+from functools import reduce
 from pathlib import Path
 from typing import TYPE_CHECKING, Any
 from urllib.parse import urlparse
 
+import pyarrow as pa
```

Comment:
This causes problems -  you'll have to use our lazy imports:

```suggestion
from polars.dependencies import pyarrow as pa
```